test/immediate_sync_test.cpp

```diff
@@ -88,3 +88,90 @@ TEST_F(ManagerTest, testDataChangeInFile)
+
+    extData::MockExternalDataIFaces* mockExtDataIfaces =
+        dynamic_cast<extData::MockExternalDataIFaces*>(extDataIface.get());
+
+    ON_CALL(*mockExtDataIfaces, fetchBMCRedundancyMgrProps())
+        // NOLINTNEXTLINE
+        .WillByDefault([&mockExtDataIfaces]() -> sdbusplus::async::task<> {
+        mockExtDataIfaces->setBMCRole(extData::BMCRole::Active);
+        co_return;
+    });
+
+    EXPECT_CALL(*mockExtDataIfaces, fetchSiblingBmcIP())
+        // NOLINTNEXTLINE
+        .WillRepeatedly([]() -> sdbusplus::async::task<> { co_return; });
+
+    EXPECT_CALL(*mockExtDataIfaces, fetchRbmcCredentials())
+        // NOLINTNEXTLINE
+        .WillRepeatedly([]() -> sdbusplus::async::task<> { co_return; });
+
+    nlohmann::json jsonData = {
+        {"Files",
+         {{{"Path", ManagerTest::tmpDataSyncDataDir.string() + "/srcFile2"},
+           {"DestinationPath",
+            ManagerTest::tmpDataSyncDataDir.string() + "/destFile2"},
+           {"Description", "File to test immediate sync upon data write"},
+           {"SyncDirection", "Active2Passive"},
+           {"SyncType", "Immediate"}}}}};
+
+    std::string srcPath{jsonData["Files"][0]["Path"]};
+    std::string destPath{jsonData["Files"][0]["DestinationPath"]};
+
+    writeConfig(jsonData);
+    sdbusplus::async::context ctx;
+
+    std::string data{"Src: Initial Data\n"};
+    // Just for create path in the dest to watch for expectation check
+    std::string destData{"Dest: Initial Data\n"};
+    ManagerTest::writeData(srcPath, data);
+    ManagerTest::writeData(destPath, destData);
+    ASSERT_EQ(ManagerTest::readData(srcPath), data);
+    ASSERT_EQ(ManagerTest::readData(destPath), destData);
+
+    data_sync::Manager manager{ctx, std::move(extDataIface),
+                               ManagerTest::dataSyncCfgDir};
+    manager.setDisableSyncStatus(true); // disabled the sync events
+
+    EXPECT_NE(ManagerTest::readData(destPath), data)
+        << "The data should not match because the manager is spawned and"
+        << " sync is disabled.";
+
+    std::string dataToWrite{"Data is modified"};
+    std::string dataToStopEvent{"Close spawned inotify event."};
+
+    // write data to src path to create inotify event so that the spawned
+    // process is closed.
+    ManagerTest::writeData(srcPath, dataToStopEvent);
+
+    ctx.spawn(sdbusplus::async::sleep_for(ctx, 1s) |
+              sdbusplus::async::execution::then(
+                  [&destPath, &dataToStopEvent, &manager]() {
+        EXPECT_NE(ManagerTest::readData(destPath), dataToStopEvent)
+            << "The data should not match as sync is disabled even though"
+            << " sync should take place at every data change.";
+        manager.setDisableSyncStatus(false); // Trigger the sync events
+    }));
+
+    // Write data after 2s so that the background sync events will be ready
+    // to catch.
+    ctx.spawn(
+        sdbusplus::async::sleep_for(ctx, 2s) |
```

Comment:
Could you confirm if 2 seconds and above 1 second are necessary? I’d like to reduce the sleep interval, if possible, to decrease the overall unit test execution time.